### PR TITLE
Handle records in documentation comments in IDE side

### PR DESCRIPTION
**Pull request by Youssef1313**
Addresses part of https://github.com/dotnet/roslyn/issues/52663.

The thing left for that issue is:

> IDE may choose to adjust inheritance logic so that properties show some information from associated parameter

@jcouv Please add this to record structs test plan. Thanks.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs [hunk 1]
```diff
@@ -270,6 +270,7 @@ public static bool IsEmbeddedStatementOwner([NotNullWhen(returnValue: true)] thi
                 SyntaxKind.ParenthesizedLambdaExpression => ((ParenthesizedLambdaExpressionSyntax)declaration).ParameterList,
                 SyntaxKind.LocalFunctionStatement => ((LocalFunctionStatementSyntax)declaration).ParameterList,
                 SyntaxKind.AnonymousMethodExpression => ((AnonymousMethodExpressionSyntax)declaration).ParameterList,
+                SyntaxKind.RecordDeclaration => ((RecordDeclarationSyntax)declaration).ParameterList,
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs:273]: note to @jcouv to have this in the record-struct branch.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Is this ready to merge?

**Comment by Youssef1313** [comment]
Pinging @CyrusNajmabadi.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**